test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1

```diff
@@ -759,6 +759,27 @@ ConstructorTestClass(int i, bool b)
         $res.CompletionMatches.CompletionText | Should -Contain "-Path"
     }
 
+    it 'Should find the closest positional parameter match' {
+        $TestString = @'
+function Verb-Noun
+{
+    Param
+    (
+        [Parameter(Position = 0)]
+        [string]
+        $Param1,
+        [Parameter(Position = 1)]
+        [System.Management.Automation.ActionPreference]
+        $Param2
+    )
+}
+Verb-Noun -Param1 Hello ^
```

Comment:
What about more complex test like (expect Value3 in parameter position 3)
```Verb-Noun -ParamPos0 Value0 ParamPos1Value1 -ParamPos2 Value2 ^```

Reply:
Is it necessary? The logic would be the same as in the previous test: It takes the next available positional parameter. It doesn't really matter if it's the first, third or even the 30th parameter position it's looking for.

Reply:
The code is too big to observe its logic in its entirety, and the test I'm asking about is a case I thought it might not work.

Reply:
That's fair. Here's how it works: The target position is found elsewhere and is based on how many positionally bound parameters have been specified, in the above example it would look for the second positional parameter (position 1) because one other positional parameter has been found.  
The loop I've modified loops over the unbound parameters and parametersets that are valid based on the currently bound parameters, here it looks for any positional parameters with a position greater than or equal to the target location. It then saves whichever parameter has the lowest position and binds the argument to that.  
Because it's looping over unbound parameters all the previous parameters "ParamPos0", "ParamPos1" and "ParamPos2" are automatically filtered out so the next available parameter would be "ParamPos3".